### ISSUE TITLE
Unexpected operator from 1.0.25-prerelease-00196/lib/init-tools.sh while doing cross build

### ISSUE THREAD

**Issue by leemgs** [dotnet/coreclr#3681]
Unexpected operator error happens from 1.0.25-prerelease-00196/lib/init-tools.sh while doing cross build. Who is in charge of the Microsoft.DotNet.BuildTools  (version 1.0.25-prerelease-00196)? In case of that, does Jenkins-based CI (Continuous Integration) system cannot currently handle the issues? 
### Build environment:

`lgs@u1404:/work/dotnet/arm-32/coreclr$ time ROOTFS_DIR=/work/dotnet/rootfs-coreclr/arm  ./build.sh arm debug verbose clean cross`

#### Error messages:
 * https://www.myget.org/F/dotnet-buildtools/
```
                       . . . . OMISSION . . . . .
Feeds used:
    https://www.myget.org/F/dotnet-buildtools/
Installed:
    1 package(s) to /work/dotnet/arm-32/coreclr/Tools/1.0.25-prerelease-00196/project.json
/work/dotnet/arm-32/coreclr/packages/Microsoft.DotNet.BuildTools/1.0.25-prerelease-00196/lib/init-tools.sh: 39: /work/dotnet/arm-32/coreclr/packages/Microsoft.DotNet.BuildTools/1.0.25-prerelease-00196/lib/init-tools.sh: source: not found
/work/dotnet/arm-32/coreclr/packages/Microsoft.DotNet.BuildTools/1.0.25-prerelease-00196/lib/init-tools.sh: 40: [: unexpected operator
/work/dotnet/arm-32/coreclr/packages/Microsoft.DotNet.BuildTools/1.0.25-prerelease-00196/lib/init-tools.sh: 42: [: unexpected operator
/work/dotnet/arm-32/coreclr/packages/Microsoft.DotNet.BuildTools/1.0.25-prerelease-00196/lib/init-tools.sh: 44: [: unexpected operator
/work/dotnet/arm-32/coreclr/packages/Microsoft.DotNet.BuildTools/1.0.25-prerelease-00196/lib/init-tools.sh: 46: [: unexpected operator
Unsupported Linux distribution '' detected. Downloading ubuntu-x64 tools.
log  : Restoring packages for /work/dotnet/arm-32/coreclr/packages/Microsoft.DotNet.BuildTools/1.0.25-prerelease-00196/lib/tool-runtime/project.json...
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.reflection.metadata/index.json
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.reflection.metadata/index.json 1041ms
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.reflection.metadata/1.3.0-rc2-23907/system.reflection.metadata.1.3.0-rc2-23907.nupkg
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.reflection.metadata/index.json
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.reflection.metadata/1.3.0-rc2-23907/system.reflection.metadata.1.3.0-rc2-23907.nupkg 1031ms
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.reflection.metadata/index.json 779ms
log  : Installing System.Reflection.Metadata 1.3.0-rc2-23907.
                       . . . . OMISSION . . . . .

```

### Related function:
 * file name: packages/Microsoft.DotNet.BuildTools/1.0.25-prerelease-00196/lib/init-tools.sh
 * location: Line 40 ~ 46
```

     32 OSName=$(uname -s)
     33 case $OSName in
     34     Darwin)
     35         __PUBLISH_RID=osx.10.10-x64
     36         ;;
     37 
     38     Linux)
     39         source /etc/os-release
     40         if [ "$ID" == "centos" ]; then
     41             __PUBLISH_RID=centos.7-x64
     42         elif [ "$ID" == "rhel" ]; then
     43             __PUBLISH_RID=rhel.7-x64
     44         elif [ "$ID" == "ubuntu" ]; then
     45             __PUBLISH_RID=ubuntu.14.04-x64
     46         elif [ "$ID" == "debian" ]; then
     47             __PUBLISH_RID=debian.8.2-x64
     48         else
     49             echo "Unsupported Linux distribution '$ID' detected. Downloading ubuntu-x64 tools."
     50             __PUBLISH_RID=ubuntu.14.04-x64
     51         fi
     52         ;;
     53 

```

**Comment by leemgs** [comment]
\cc @jkotas , @AndyAyersMS, @adityamandaleeka, @kyulee1 , @myungjoo 

We are still getting the `"/Microsoft.DotNet.BuildTools/1.0.25-prerelease-00196/lib/init-tools.sh: 40: [: unexpected operator ..."` messages during build time. It's strange.  It seems that CI (Continuous Integration) System could not catch the error because this error was made by the server package such as Microsoft.DotNet.BuildTool.  Could anyone forward this issue to the person in charge of this error?

**Comment by jkotas** [comment]
The master copy of the init-tools.sh script lives at https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/init-tools.sh

**Comment by leemgs** [comment]
> init-tools.sh: source: not found

Thank you for sharing info.  I will submit a PR to `./dotnet/buildtools/` community to fix this issue. **source** is a bash build-in command. It is not a program - like ls or grep. Like the below example, on up-to-date Ubuntu distribution, /bin/sh calls /bin/dash. Traditionally, `/bin/sh` called `/bin/bash` for supporting compatibility mode. 

```
ls -l `which sh`
invain@target:~$ ls -l `which sh`
lrwxrwxrwx 1 root root 4 12월 21 17:01 /bin/sh -> dash
invain@target:~$ 

```

So,  we have to use **cat** command instead of **source** command  as following: 

```
     38     Linux)
     40         if [ "$(cat /etc/*-release | grep -cim1 centos)" -eq 1 ]; then
     41             __PUBLISH_RID=centos.7-x64
     42         elif [ "$(cat /etc/*-release | grep -cim1 rhel)" -eq 1 ]; then
     43             __PUBLISH_RID=rhel.7-x64
     44         elif [ "$(cat /etc/*-release | grep -cim1 ubuntu)" -eq 1 ]; then
     45             __PUBLISH_RID=ubuntu.14.04-x64
     46         elif [ "$(cat /etc/*-release | grep -cim1 debian)" -eq 1 ]; then
     47             __PUBLISH_RID=debian.8.2-x64
     48         else
     49             echo "Unsupported Linux distribution '$ID' detected. Downloading ubuntu-x64 tools."
     50             __PUBLISH_RID=ubuntu.14.04-x64
     51         fi
```

**Comment by ellismg** [comment]
Fixed with #3830
